morpheus/messages/multi_inference_message.py

```diff
@@ -24,61 +24,56 @@
 from morpheus.messages.tensor_memory import TensorMemory
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(init=False)
 class InferenceMemory(TensorMemory, cpp_class=_messages.InferenceMemory):
     """
     This is a base container class for data that will be used for inference stages. This class is designed to
     hold generic tensor data in cupy arrays.
     """
 
+    def get_input(self, name: str):
+        """
+        Getter function used with DataClassProp for getting inference input from message containers derived
+        from InferenceMemory.
 
-def get_input(instance, name: str):
-    """
-    Getter function used with DataClassProp for getting inference input from message containers derived
-    from InferenceMemory.
-
-    Parameters
-    ----------
-    instance : `InferenceMemory`
-        Message container holding inputs.
-    name : str
-        Key used to do lookup in inputs dict of message container.
-
-    Returns
-    -------
-    cupy.ndarray
-        Inputs corresponding to name.
-
-    Raises
-    ------
-    AttributeError
-        If input name does not exist in message container.
-    """
-    if (name not in instance.tensors):
-        raise AttributeError
+        Parameters
+        ----------
+        name : str
+            Key used to do lookup in inputs dict of message container.
 
-    return instance.tensors[name]
+        Returns
+        -------
+        cupy.ndarray
+            Inputs corresponding to name.
 
+        Raises
+        ------
+        AttributeError
+            If input name does not exist in message container.
+        """
+        try:
+            return self.get_tensor(name)
+        except KeyError:
+            raise AttributeError
```

Comment:
You dont want to do this here. `AttributeError` should only be used in `__getattr__` and should return a `KeyError`. `AttributeError` is handled differently than other exceptions and tells the runtime to look at self.__dict__ instead.